userspace/libsinsp/cri.cpp

```diff
@@ -400,22 +401,64 @@ bool cri_interface::is_pod_sandbox(const std::string &container_id)
 	return status.ok();
 }
 
-uint32_t cri_interface::get_pod_sandbox_ip(const std::string &pod_sandbox_id)
+void cri_interface::get_pod_info_cniresult(runtime::v1alpha2::PodSandboxStatusResponse &resp, std::string &cniresult)
+{
+	Json::Value root;
+	Json::Reader reader;
+	const auto &info_it = resp.info();
+	if(reader.parse(info_it.find("info")->second, root))
+	{
+		Json::Value jvalue;
+		if (m_cri_runtime_type == CT_CONTAINERD)
+		{
+			jvalue = root["cniResult"]["Interfaces"];
+			if(!jvalue.isNull())
+			{
+				/* If applicable remove members / fields not needed for incident response. */
+				jvalue.removeMember("lo");
+				for (auto& key : jvalue.getMemberNames())
+				{
+					if (0 == strncmp(key.c_str(), "veth", 4))
+					{
+						jvalue.removeMember(key);
+					} else
+					{
+						jvalue[key].removeMember("Mac");
+						jvalue[key].removeMember("Sandbox");
+					}
+				}
+			}
+
+		} else if (m_cri_runtime_type == CT_CRIO)
+		{
+			/* CRIO container runtime does not expose cni result as parsed JSON, it's an escaped JSON string that contains list of ips without knowing the interface names. */
+			jvalue = root["runtimeSpec"]["annotations"]["io.kubernetes.cri-o.CNIResult"];
+		}
+
+		if(!jvalue.isNull())
+		{
+			Json::FastWriter fastWriter;
+			cniresult = fastWriter.write(jvalue);
+			std::string chars = "\n\\";
+			cniresult.erase(remove_if(cniresult.begin(), cniresult.end(),[&chars](const char &c) {return chars.find(c) != std::string::npos;}), cniresult.end());
+			cniresult.resize(MAX_CNIRESULT_LENGTH);
```

Comment:
1. I find the `erase(remove_if(...))` part hard to follow TBH but it looks like we're stripping all `\n`s and `\`s from the string. Do we actually need this? If the data is double-JSON-encoded as a string in JSON, the string should be properly decoded by the JSON library, no? (note: I haven't seen a sample input here so I may well be wrong)
2. `.resize()` will pad the string up to the maximum length. Do we want this or should we resize only if `cniresult.size() > MAX_CNIRESULT_LENGTH`?

Reply:
Addressed feedback. Kept original sanitization, but adjusted style a bit. Is it better this way? 

Shared some sample results here https://github.com/falcosecurity/libs/pull/708#issuecomment-1362260873, just imagine without removing the `\` in the Falco logs you would have a triple `\\\` in the CRIO case. Have done enough Big Data stuff and ETLs to highly recommend going with the consistent escaped JSON and nuke the `\n` already as well.

Reply:
Thanks @gnosek for your help cleaning this up :pray: